polyfactory/factories/pydantic_factory.py

```diff
@@ -150,13 +217,25 @@ def get_model_fields(cls) -> list["FieldMeta"]:
 
         """
         if "_fields_metadata" not in cls.__dict__:
-            cls._fields_metadata = [
-                PydanticFieldMeta.from_model_field(
-                    field,
-                    use_alias=not cls.__model__.__config__.allow_population_by_field_name,
-                )
-                for field in cls.__model__.__fields__.values()
-            ]
+            if pydantic_version == 1:
+                cls._fields_metadata = [
+                    PydanticFieldMeta.from_model_field(
+                        field,
+                        use_alias=not cls.__model__.__config__.allow_population_by_field_name,
+                    )
+                    for field in cls.__model__.__fields__.values()  # type: ignore[attr-defined]
+                ]
+            # FIXME: remove the pragma when switching to pydantic v2 permanently
```

Comment:
I see you are testing against both pydantic v1 and v2 in the ci matrix, so you could test with coverage for both and then use `coverage combine`, so that coverage of all the branches is included.